test-case-reporting/src/db_interfacing.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2020 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Build, DB, Job, TestCase, TestRun} from 'test-case-reporting';
+import {Database, TIMESTAMP_PRECISION} from './db';
+
+// Types in the DB namespace interface use snake_case instead of camelCase.
+/* eslint @typescript-eslint/camelcase: "off" */
+function getDateFromTimestamp(timestamp: number): Date {
+  const msConversionConstant: number = Math.pow(10, 3 - TIMESTAMP_PRECISION);
+  return new Date(timestamp * msConversionConstant);
+}
+
+export class TestResultsRecords {
+  constructor(private db: Database) {}
+
+  private async getDbBuild(buildNumber: string): Promise<DB.Build> {
+    return this.db<DB.Build>('builds')
+      .where('build_number', buildNumber)
+      .first();
+  }
+
+  private async getBuildWithEmptyJobs(buildNumber: string): Promise<Build> {
+    const dbBuild = await this.getDbBuild(buildNumber);
+    return {
+      commitSha: dbBuild.commit_sha,
+      buildNumber: dbBuild.build_number,
+      startedAt: getDateFromTimestamp(dbBuild.started_at),
+      jobs: [],
+    };
+  }
+
+  private makeTestCaseFromRow(row): TestCase {
+    return {
+      name: row.test_case_name,
+      createdAt: getDateFromTimestamp(row.test_case_created_at),
+    };
+  }
+
+  private makeTestRunFromRow(row): TestRun {
+    return {
+      testCase: this.makeTestCaseFromRow(row),
+      status: row.test_run_status,
+      timestamp: getDateFromTimestamp(row.test_run_timestamp),
+      durationMs: row.duration_ms,
+    };
+  }
+
+  // I need help typechecking this
```

Comment:
Add comments to the PR, rather than in code where it's easy to forget to remove them 🙃

Reply:
Whoops -- I was kind of thinking-out-loud on comments in the code and I forgot to remove that one (proving your point).

Reply:
Done.